Sources/Sentry/SentryCoreDataTracker.m

```diff
@@ -0,0 +1,83 @@
+
+#import "SentryCoreDataTracker.h"
+#import "SentryHub+Private.h"
+#import "SentryLog.h"
+#import "SentrySDK+Private.h"
+#import "SentryScope+Private.h"
+#import "SentrySpanProtocol.h"
+
+@implementation SentryCoreDataTracker
+
+- (NSArray *)managedObjectContext:(NSManagedObjectContext *)context
+              executeFetchRequest:(NSFetchRequest *)request
+                            error:(NSError **)error
+                      originalImp:(NSArray *(NS_NOESCAPE ^)(NSFetchRequest *, NSError **))original
+{
+    __block id<SentrySpan> fetchSpan;
+    [SentrySDK.currentHub.scope useSpan:^(id<SentrySpan> _Nullable span) {
+        fetchSpan = [span startChildWithOperation:SENTRY_COREDATA_FETCH_OPERATION
+                                      description:[self descriptionFromRequest:request]];
+    }];
+    NSArray *result = original(request, error);
+
+    [fetchSpan setDataValue:[NSNumber numberWithInteger:result.count] forKey:@"result_amount"];
+
+    [fetchSpan
+        finishWithStatus:error != nil ? kSentrySpanStatusInternalError : kSentrySpanStatusOk];
+
+    return result;
+}
+
+- (BOOL)managedObjectContext:(NSManagedObjectContext *)context
+                        save:(NSError **)error
+                 originalImp:(BOOL(NS_NOESCAPE ^)(NSError **))original
+{
+
+    __block id<SentrySpan> fetchSpan = nil;
+    if (context.hasChanges) {
+        [SentrySDK.currentHub.scope useSpan:^(id<SentrySpan> _Nullable span) {
+            fetchSpan = [span startChildWithOperation:SENTRY_COREDATA_SAVE_OPERATION
+                                          description:@"Saving Database"];
```

Comment:
Ideally this would have the name of the dataset or other data that will help the user understand what's being saved.
If not something out of the box, possibly something that we can document users: "Set the name in this property to see it in the span description"